src/shapes/object.class.ts

```diff
@@ -1,2049 +1,2023 @@
+   * 0 is no corner, or 'mt', 'ml', 'mtr' etc..
+   * It should be private, but there is no harm in using it as
+   * a read-only property.
+   * @type number|string|any
+   * @default 0
+   */
+  __corner: number | string;
+
+  /**
+   * Determines if the fill or the stroke is drawn first (one of "fill" or "stroke")
+   * @type String
+   * @default
+   */
+  paintFirst: 'fill' | 'stroke';
+
+  /**
+   * When 'down', object is set to active on mousedown/touchstart
+   * When 'up', object is set to active on mouseup/touchend
+   * Experimental. Let's see if this breaks anything before supporting officially
+   * @private
+   * since 4.4.0
+   * @type String
+   * @default 'down'
+   */
+  activeOn: 'down' | 'up';
+
+  /**
+   * List of properties to consider when checking if state
+   * of an object is changed (fabric.Object#hasStateChanged)
+   * as well as for history (undo/redo) purposes
+   * @type Array
+   */
+  stateProperties: string[];
+
+  /**
+   * List of properties to consider when checking if cache needs refresh
+   * Those properties are checked by statefullCache ON ( or lazy mode if we want ) or from single
+   * calls to Object.set(key, value). If the key is in this list, the object is marked as dirty
+   * and refreshed at the next render
+   * @type Array
+   */
+  cacheProperties: string[];
+
+  /**
+   * List of properties to consider for animating colors.
+   * @type String[]
+   */
+  colorProperties: string[];
+
+  /**
+   * a fabricObject that, without stroke define a clipping area with their shape. filled in black
+   * the clipPath object gets used when the object has rendered, and the context is placed in the center
+   * of the object cacheCanvas.
+   * If you want 0,0 of a clipPath to align with an object center, use clipPath.originX/Y to 'center'
+   * @type fabric.Object
+   */
+  clipPath: FabricObject | undefined;
+
+  /**
+   * Meaningful ONLY when the object is used as clipPath.
+   * if true, the clipPath will make the object clip to the outside of the clipPath
+   * since 2.4.0
+   * @type boolean
+   * @default false
+   */
+  inverted: boolean;
+
+  /**
+   * Meaningful ONLY when the object is used as clipPath.
+   * if true, the clipPath will have its top and left relative to canvas, and will
+   * not be influenced by the object transform. This will make the clipPath relative
+   * to the canvas, but clipping just a particular object.
+   * WARNING this is beta, this feature may change or be renamed.
+   * since 2.4.0
+   * @type boolean
+   * @default false
+   */
+  absolutePositioned: boolean;
+
+  canvas?: StaticCanvas | Canvas;
+  /**
+   * Unique id used internally when creating SVG elements
+   * @static
+   * @memberOf fabric.Object
+   * @type Number
+   */
+  static __uid = 0;
+
+  /**
+   * Constructor
+   * @param {Object} [options] Options object
+   */
+  constructor(options: FabricObjectDefaultValues & Record<string, unknown>) {
```

Comment:
```suggestion
  constructor(options: typeof FabricObjectDefaultValues & Record<string, unknown>) {
```

Reply:
yes this is wrong for now, and will stay an any, for now till we can clearly separate options from internal values that are optionals and **should not be used as options** ( for example _cacheCanvas is a valid value but absolutely not an option )